pkg/cloudprovider/instancetypes.go

```diff
@@ -69,19 +69,12 @@ type InstanceTypeProvider struct {
 }
 
 func NewInstanceTypeProvider(ctx context.Context, sess *session.Session, ec2api ec2iface.EC2API, subnetProvider *subnet.Provider,
-	unavailableOfferingsCache *awscache.UnavailableOfferings, startAsync <-chan struct{}) *InstanceTypeProvider {
+	unavailableOfferingsCache *awscache.UnavailableOfferings, pricingProvider *pricing.Provider, startAsync <-chan struct{}) *InstanceTypeProvider {
```

Comment:
```suggestion
	unavailableOfferingsCache *awscache.UnavailableOfferings, pricingProvider *pricing.Provider) *InstanceTypeProvider {
```

I don't think we need this anymore